roles/pre_tasks/tasks/main.yml

```diff
@@ -1,10 +1,23 @@
 ---
 
 # Install absolutely necessary packages
+
+- name: Check if the absolutely necessary packages are installed
+  shell: "type {{ item }}"
+  with_items: "{{ tower_pre_tasks_packages }}"
+  register: type_packages
+  ignore_errors: yes
+
 - name: "Install absolutely necessary packages"
   yum:
-    name: "{{ tower_pre_tasks_packages }}"
-    state: present
+    name: "{{ package_item.item }}"
+    state: latest
+  ignore_errors: true
+  loop: "{{ type_packages.results }}"
+  loop_control:
+    loop_var: package_item
```

Comment:
To be consistent, you should also rename the loop_var on the previous loop, or rename none of them.

Reply:
Updated, Agree with changes, updated this var like oasis